melee/SSBMEnv.py

```diff
@@ -396,27 +440,20 @@ def state_np_to_list(self, dictionary):
                 self.state_np_to_list(dictionary[entry])
     
     def dump_state(self):
-        if self.statedump_prefix is not None:
-            if self.statedump_n > 0:
-                with open(f"./{self.statedump_prefix}_{self.statedump_n}.txt", "w") as f:
-                    for entry in self.state_data:
-                        self.state_np_to_list(entry)
-                    f.write(json.dumps(self.state_data))
+        if self.dump_states and self.state_data:
+            with open(os.path.join(self.statedump_dir, f"{self.statedump_prefix}_{self.statedump_n}.txt"), "w") as f:
+                for entry in self.state_data:
+                    self.state_np_to_list(entry)
+                f.write(json.dumps(self.state_data))
             self.statedump_n += 1
         self.state_data = []
 
-    def reset(self):    # TODO: should reset state to initial state, how to do this?
-        self.dump_state()
```

Comment:
So we just want the caller to dump state instead of doing it in reset? I'll need to double check but this might be in __main__ already

Reply:
Nope, it's still called automatically in `reset` just like before (see line 451). These lines were removed because the `reset` function ended up appearing twice when everything was merged together earlier today

Reply:
oh got it. Not sure if what I'm doing in the __main__ to always dump state on program end or Ctrl+C interrupt is necessary then. But at least it didn't seem to be overwriting data when I tried it so 🤷